shacl/index.bs

```diff
@@ -0,0 +1,83 @@
+<pre class='metadata'>
+Title: Hydra-SHACL Interoperability
+Shortname: hashi
+Level: 1
+Status: LD
+Group: HydraCG
+URL: https://hydracg.github.io/extensions/shacl
+Editor: Tomasz Pluskiewicz, Zazuko GmbH https://zazuko.com/, https://t-code.pl
+Abstract: This document provides additional terms and examples of using SHACL together with Hydra to describe API's hyperemedia
+Markup Shorthands: markdown yes
+</pre>
+
+<pre class=biblio>
+{
+	"hydra": {
+		"authors": [
+			"Markus Lanthaler"
+		],
+		"href": "http://www.hydra-cg.com/spec/latest/core/",
+		"title": "Hydra Core Vocabulary",
+		"status": "UD",
+		"publisher": "W3C"
+	}
+}
+</pre>
+
+# Introduction # {#intro}
+
+[[!Hydra|Hydra Core]] vocabulary does define a handful of data modeling terms such as `hydra:Class` and `hydra:supportedProperty` which can be used to build basic APIs. However, it was never a goal of Hydra Core to become a full-blown modeling language, given available alternatives.
+
+One such alternative is [[!SHACL]], which is defined by its authors as "a language for validating RDF graphs against a set of conditions", yet its usefulness goes way beyond just data validation. See the [[shacl-ucr|SHACL Use Cases and Requirements]] note for various ideas for employing [[SHACL]].
+
+In this document we propose various ways in which [[SHACL]] can be used by both Hydra API publishers and consumers to facilitate client-server interaction.
+
+# Conventions # {#conventions}
+
+The base namespace of this extension and its unique identifier is `http://www.w3.org/ns/hydra/shacl#`.
+
+The preferred prefix is `hashi:`.
+
+#  Announcing the extension # {#announcing}
+
+Conforming with the [[Hydra#extensions|Extensions]] section of [[Hydra]], an API MUST announce the use of an extension by adding it to its `hydra:ApiDocumentation`:
+
+<div class='example'>
```

Comment:
This mixture of markdown-like plain text and pre-formatted HTML looks poor. Is it possible to use something closer to MD code format like ``` notation ?

Reply:
You cannot add a CSS classes to fenced codeblocks. It is possible to [include files](https://tabatkins.github.io/bikeshed/#including-code) instead it it would make you feel better but I doubt it makes sense unless they are slightly bigger (with the option to also include only part, which may work quite nice for the SHACL shape)

Reply:
That's why the question on BS files. MD would look cleaner.

Reply:
But does not produce a nice HTML spec like Bikeshed does. At least not without more opinionated tooling anyway.

Check the artefact produced by the GitHub action. The master branch will publish the rendered HTML as gh-pages